lnwallet/chanfunding/coin_select_test.go

```diff
@@ -42,6 +50,107 @@ func fundingFee(feeRate chainfee.SatPerKWeight, numInput int, // nolint:unparam
 	return feeRate.FeeForWeight(totalWeight)
 }
 
+// TestCalculateFees tests that the helper function to calculate the fees
+// both with and without applying a change output is done correctly for
+// (N)P2WKH inputs, and should raise an error otherwise.
+func TestCalculateFees(t *testing.T) {
+	t.Parallel()
+
+	const feeRate = chainfee.SatPerKWeight(1000)
+
+	type testCase struct {
+		name  string
+		utxos []Coin
+
+		expectedFeeNoChange   btcutil.Amount
+		expectedFeeWithChange btcutil.Amount
+		expectErr             bool
+	}
+
+	testCases := []testCase{
+		{
+			name: "one P2WKH input",
+			utxos: []Coin{
+				{
+					TxOut: wire.TxOut{
+						PkScript: p2wkhScript,
+						Value:    1,
+					},
+				},
+			},
+
+			expectedFeeNoChange:   487,
+			expectedFeeWithChange: 611,
+			expectErr:             false,
+		},
+
+		{
+			name: "one NP2WKH input",
+			utxos: []Coin{
+				{
+					TxOut: wire.TxOut{
+						PkScript: np2wkhScript,
+						Value:    1,
+					},
+				},
+			},
+
+			expectedFeeNoChange:   579,
+			expectedFeeWithChange: 703,
+			expectErr:             false,
+		},
+
+		{
+			name: "not supported P2KH input",
+			utxos: []Coin{
+				{
+					TxOut: wire.TxOut{
+						PkScript: p2khScript,
+						Value:    1,
+					},
+				},
+			},
+
+			expectErr: true,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			feeNoChange, feeWithChange, err := calculateFees(
+				test.utxos, feeRate,
+			)
+			if !test.expectErr && err != nil {
+				t.Fatalf(err.Error())
+			}
+
+			if test.expectErr && err == nil {
```

Comment:
could add an `expectedError` field to the test and `require.Equal`, you'd only need to define an `errUnsupportedInput` error since that's the only error we're testing for. Simplifies this a bit.